tracker/progress.go

```diff
@@ -127,13 +139,15 @@ func (pr *Progress) BecomeProbe() {
 	} else {
 		pr.ResetState(StateProbe)
 		pr.Next = pr.Match + 1
+		pr.pendingCommit = min(pr.pendingCommit, pr.Next-1)
```

Comment:
Why does not execute this for the if case `if pr.State == StateSnapshot`?

Reply:
Bug. Moved this line to the right place.

Reply:
Actually, `pendingCommit` is not well defined in `StateSnapshot`. We should probably set it to `PendingSnapshot`.

Reply:
Done. This further reduces the amount of unnecessary messages.